Fix connect-publish action

The [new action](https://github.com/The-Strategy-Unit/nhp_inputs_report_app/blob/main/.github/workflows/connect-publish.yaml) was triggered correctly on release of v0.8.0, but [failed](https://github.com/The-Strategy-Unit/nhp_inputs_report_app/actions/runs/17124679395) when creating the manifest:

```
Run rsconnect::writeManifest()
Error: Error in loadNamespace(x) : there is no package called ‘rsconnect’
Calls: loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
Error: Process completed with exit code 1.
```

As in the `rsconnect::writeManifest()` call can't be made because {rsconnect} isn't installed into the environment? So perhaps {rsconnect} needs adding to `DESCRIPTION` and then [the setup-renv step](https://github.com/r-lib/actions/tree/v2-branch/setup-renv) is needed in the action?

Or can we just install {rsconnect} in its own step so that the subsequent `rsconnect::writeManifest()` step will work?

```
- name: Install rsconnect
  shell: Rscript {0}
  run: |
    install.packages("rsconnect")
```

--- a/.github/workflows/connect-publish.yaml
+++ b/.github/workflows/connect-publish.yaml
@@ -14,7 +14,8 @@ jobs:
           r-version: latest
           use-public-rspm: true
 
-      - uses: r-lib/actions/setup-renv@v2
+      - name: Install dependencies
+        uses: r-lib/actions/setup-r-dependencies@v2
 
       - name: Create manifest.json
         shell: Rscript {0}

--- a/.github/workflows/connect-publish.yaml
+++ b/.github/workflows/connect-publish.yaml
@@ -14,6 +14,8 @@ jobs:
           r-version: latest
           use-public-rspm: true
 
+      - uses: r-lib/actions/setup-renv@v2
+
       - name: Create manifest.json
         shell: Rscript {0}
         run: |

--- a/.github/workflows/connect-publish.yaml
+++ b/.github/workflows/connect-publish.yaml
@@ -16,6 +16,9 @@ jobs:
 
       - name: Install dependencies
         uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+            extra-packages: |
+                any::rsconnect
 
       - name: Create manifest.json
         shell: Rscript {0}

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,9 +1,10 @@
 Package: nhp.inputs.report.app
 Title: NHP Mitigator Comparison App
-Version: 0.7.2
+Version: 0.8.1
 Authors@R: c(
     person("Matt", "Dray", , "matt.dray@nhs.net", role = c("cre", "aut")),
     person("Craig", "Parylo", role = "aut"),
+    person("Claire", "Welsh", role = "ctb"),
     person("Ozayr", "Mohammed", role = "ctb"),
     person("Anya", "Ferguson", role = "ctb")
   )
@@ -52,6 +53,7 @@ Imports:
     withr,
     yaml
 Suggests: 
+    rsconnect,
     testthat (>= 3.0.0)
 Remotes: 
     dreamRs/datamods

